### PR TITLE
GitHub actions 21.04

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: .github/workflows/ci-python.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+      - name: Install poetry and dependencies
+        uses: greenbone/actions/poetry@v1
+      - name: Calculate and upload coverage to codecov.io
+        uses: greenbone/actions/coverage-python@v1
+        with:
+          test-command: -m unittest
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:63]:
```suggestion
          test-command: -m unittest

```

File: .github/workflows/ci-python.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  codecov:
+    name: Upload coverage to codecov.io
+    needs: test
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:57]:
```suggestion
          python-version: 3.8
```

File: .github/workflows/ci-python.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install poetry and dependencies
+        uses: greenbone/actions/poetry@v1
+      - name: Check with black, pylint and pontos.version
+        uses: greenbone/actions/lint-python@v1
+        with:
+          packages: ospd_openvas
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:28]:
```suggestion
          packages: ospd_openvas tests
```

File: .github/workflows/ci-python.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    branches: [ master, ospd-openvas-21.04, ospd-openvas-20.08 ]
+
+jobs:
+  linting:
+    name: Linting
+    runs-on: 'ubuntu-latest'
+    strategy:
+        matrix:
+          python-version:
+            - 3.7
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:16]:
```suggestion
            - 3.7
            - 3.8
            - 3.9
```

File: .github/workflows/ci-python.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          packages: ospd_openvas
+
+  test:
+    name: Run all tests
+    runs-on: 'ubuntu-latest'
+    strategy:
+        matrix:
+          python-version:
+            - 3.7
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:36]:
```suggestion
            - 3.7
            - 3.8
            - 3.9
```

File: .github/workflows/codeql-analysis-python.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```
Comment (y0urself) [inline .github/workflows/codeql-analysis-python.yml:35]:
```suggestion
      uses: github/codeql-action/analyze@v1

```

File: .github/workflows/deploy-pypi.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        python -m pip install --upgrade pip
+        python -m pip install --upgrade poetry
+        python -m pip install --upgrade twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      run: |
+        poetry build
+        twine upload dist/*
```
Comment (y0urself) [inline .github/workflows/deploy-pypi.yml:27]:
```suggestion
        twine upload dist/*

```

File: .github/workflows/deploy-pypi.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
Comment (y0urself) [inline .github/workflows/deploy-pypi.yml:15]:
```suggestion
        python-version: 3.8
```

**Review by y0urself — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-python.yml (hunk 1 above)
Comment (bjoernricks) [reply .github/workflows/ci-python.yml:63]: btw. -m unittest is the default and can therefore be removed

Comment (y0urself) [inline .github/workflows/ci-python.yml:63]:
```suggestion

```

**Review by y0urself — CHANGES_REQUESTED** [review]

**Review by y0urself — DISMISSED** [review]

File: .github/workflows/ci-python.yml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install poetry and dependencies
+        uses: greenbone/actions/poetry@v1
+      - name: Check with black, pylint and pontos.version
+        uses: greenbone/actions/lint-python@v1
+        with:
+          packages: ospd_openvas tests
```
Comment (y0urself) [inline .github/workflows/ci-python.yml:30]:
```suggestion
          packages: ospd_openvas
```

**Review by y0urself — APPROVED** [review]

**Pull request merged**